pkg/services/alerting/rule.go

```diff
@@ -207,6 +209,32 @@ func NewRuleFromDBAlert(ctx context.Context, store AlertStore, ruleDef *models.A
 		return nil, ValidationError{Reason: "Alert is missing conditions"}
 	}
 
+	if ruleDef.EvalData != nil {
```

Comment:
Can you give me a brief explanation of what's happening here? In which cases would `EvalData` not be empty and we are we setting  `EvalMatches`?

Reply:
- This field is of type `*simplejson.Json`. Since it's a pointer, calling a method like `.Get()` could result in panics. I've run into this while trying to update alerts, thus the check to see whether it's nil.
- If it's non-nil but empty, then the `.MustArray()` call will return an empty slice, so the range loop won't execute. This happens when the alert rule goes from `Alerting` to `OK` and the database gets updated with empty evaluation matches.
- If it's non-nil and not empty, like every time we fetch all alerts from the database, then the eval matches get translated from `simplejson.Json` to `[]*EvalMatch`.

So, every time we query the database we get back evaluation data. We were not using this before, but now we are parsing that JSON into a slice of the correct type to be used later in the `EvalContext`.